src/Raven.Client/Documents/Operations/Attachments/GetAttachmentsOperation.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Attachments;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Client.Json.Converters;
+using Raven.Client.Util;
+using Sparrow;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.Attachments
+{
+    public class GetAttachmentsOperation : IOperation<IEnumerator<AttachmentEnumeratorResult>>
+    {
+        private readonly AttachmentType _type;
+        private readonly IEnumerable<AttachmentRequest> _attachments;
+
+        public GetAttachmentsOperation(IEnumerable<AttachmentRequest> attachments, AttachmentType type)
+        {
+            _type = type;
+            _attachments = attachments;
+        }
+
+        public RavenCommand<IEnumerator<AttachmentEnumeratorResult>> GetCommand(IDocumentStore store, DocumentConventions conventions, JsonOperationContext context, HttpCache cache)
+        {
+            return new GetAttachmentsCommand(context, _attachments, _type);
+        }
+
+        internal class GetAttachmentsCommand : RavenCommand<IEnumerator<AttachmentEnumeratorResult>>
+        {
+            private readonly JsonOperationContext _context;
+            private readonly AttachmentType _type;
+            internal IEnumerable<AttachmentRequest> Attachments { get; }
+            internal List<AttachmentDetails> AttachmentsMetadata { get; } = new List<AttachmentDetails>();
```

Comment:
Are those used outside of this command?

Reply:
yes I use them for nameof() in attachmentHandler & operation.